docs/developer/web.rst

```diff
@@ -0,0 +1,224 @@
+error object in the response body. The error object **should** contain a unique
+error code (that can be looked up for more details), an error message and
+possibly a detailed description:
+
+.. code-block:: JSON
+
+    {
+        "code": 1234,
+        "message": "Something went wrong.",
+        "description": "What exactly went wrong."
+    }
+
+Furthermore, every error response **must** use a meaningful `HTTP status code
+<https://tools.ietf.org/html/rfc7231#section-6.1>`_.
+The status code **may** also be part of the error object in the response body.
+
+
+5. URL structure
+----------------
+
+URLs **should** be RESTful in the sense that they represent logical resources
+or collections of resources. The plural form should be used as endpoint names.
+Whenever possible, a single word should be used to describe resources. Where
+compound words cannot be avoided, they are built using **dashes**. For complex
+queries (filtering, sorting) query parameters **should** be used.
+
+
+- URL for the entire collection of contexts:
+    ``/contexts``
+
+- URL for a specific context:
+    ``/contexts/{id}``
+
+
+5.1 Relations
+~~~~~~~~~~~~~
+A relation between resources **must** be represented in the URL structure if
+the related resources only exist in the context of the relation. For example,
+if in a situation where a file has primary key composed of a bucket identifier
+and a file identifier, the URLs would be structured as followed:
+
+- URL for all files inside the bucket:
+    ``/buckets/{bucket}/files``
+
+- URL for a specific file in a given bucket:
+    ``/buckets/{bucket}/files/{file}``
+
+If a related resource does exist independently of the related resource, relations
+are still represented in the URL structure, however individual resources are
+accessed independently.
+
+- URL for all executions related to a given context:
+    ``/contexts/{id}/executions``
+
+All relations of a resource **should** be made available following such patterns.
+
+- ``/projects/{id}/creator`` - single resource can be a ``301`` redirect to the
+  resource base URL (e.g. ``Location: /users/1234``).
+- ``/projects/{id}/buckets``
+
+In contrast, a specific resource is accessed independently of the relations which
+have been followed.
+
+- URL for a specific execution whose id was retrieved trough a context:
+    ``/executions/{id}``
+
+To make following relations more convenient, links to the related resources **should**
+be included in the response body when a given relation is queried. In case of a single
+resource response the links **may** also be included in the response `link header
+<https://tools.ietf.org/html/rfc5988#page-6>`_.
+
+
+6. Query Parameters
+-------------------
+
+6.1 Filtering
+~~~~~~~~~~~~~
+
+Unique query parameters **must** be used for each field that allows filtering.
+
+- Retrieving all executions which are currently running:
+    ``GET /executions?status=running``
```

Comment:
It will be important to document all the accepted parameters. Or would those be dynamic according to the model of the resource (allowing to filter on all attributes)?

Reply:
> used for each field that allows filtering

It should be a documented list of attributes that allow filtering.